range.c

```diff
@@ -194,15 +212,26 @@ recursive_eql(VALUE range, VALUE obj, int recur)
 
 /*
  *  call-seq:
- *     rng.eql?(obj)    -> true or false
+ *    eql?(other) -> true or false
+ *
+ *  Returns +true+ if and only if:
  *
- *  Returns <code>true</code> only if +obj+ is a Range, has equivalent
- *  begin and end items (by comparing them with <code>eql?</code>),
- *  and has the same #exclude_end? setting as the range.
+ *  - +other+ is a range.
+ *  - <tt>other.begin eql? self.begin</tt>.
+ *  - <tt>other.end eql? self.end</tt>.
+ *  - <tt>other.exclude_end? == self.include_end?</tt>.
```

Comment:
It would be useful to show an example where Range#== behavior differs from Range#eql? behavior, such as:

```ruby
(1..2) == (1..2.0)
# => true
(1..2).eql? (1..2.0)
# => false
```

Reply:
Added note in both places, plus Related.